eq_cir_management_ui/__init__.py

```diff
@@ -0,0 +1,134 @@
+"""Flask application factory for the EQ CIR Management UI."""
+
+import json
+import logging
+import os
+from pathlib import Path
+
+from flask import Flask
+from flask_talisman import Talisman
+from jinja2 import ChainableUndefined, FileSystemLoader
+from semver.version import Version
+
+from eq_cir_management_ui.config.config import DefaultConfig
+from eq_cir_management_ui.errors.routes import errors_blueprint
+from eq_cir_management_ui.main.routes import main_blueprint
+from eq_cir_management_ui.utils.routes import utils_blueprint
+
+logger = logging.getLogger()
+
+talisman = Talisman()
+
+
+def create_app(app_config: type[DefaultConfig]) -> Flask:
+    """Flask application factory, used to isolate the instance of the Flask application.
+    See https://flask.palletsprojects.com/en/2.2.x/patterns/appfactories/ .
+    """
+    app = Flask(__name__)
+
+    app.config.from_object(app_config)
+
+    app.register_blueprint(main_blueprint)
+    app.register_blueprint(errors_blueprint)
+    app.register_blueprint(utils_blueprint)
+
+    jinja_config(app)
+    design_system_config()
+    configure_secure_headers(app)
+
+    return app
+
+
+def env_override(value: str, key: str) -> str:
+    """Jinja filter to override a value with an environment variable if it exists.
+    :param value: The default value to use if the environment variable is not set.
+    :param key: The name of the environment variable to check.
+    :return: The value of the environment variable if it exists, otherwise the default value.
+    """
+    return os.getenv(key, value)
+
+
+def jinja_config(app: Flask) -> None:
+    """Configuration for the Flask Jinja2 component. Here we provide a custom loader,
+    so we can load from an array of sources.
+
+    :param app: The Flask application.
+    """
+    # loader for local templates and design system component templates
+    file_system_loader = FileSystemLoader([Path("./node_modules/@ons/design-system"), Path("./templates")])
+
+    app.jinja_loader = file_system_loader
+    app.jinja_env.undefined = ChainableUndefined
+    app.jinja_env.filters["env_override"] = env_override
+
+    # Clean up white space.
+    app.jinja_env.trim_blocks = True
+    app.jinja_env.lstrip_blocks = True
+
+
+def design_system_config() -> None:
+    """Set the version of the design system to an environment variable and add an
+    environment variable filter so environment variables can be read from within
+    Jinja. This enables the design system version to be defined once within the
+    package.json file and then reused throughout the application. Primarily to
+    declare the CSS version to use.
+    """
+    with open(Path("./package.json"), encoding="utf-8") as file:
+        package_json = json.load(file)
+
+        if "dependencies" in package_json:
```

Comment:
Might be slightly cleaner and more readable by doing something like this and then checking the var rather than the try/except and nested if:
```suggestion
        design_system_version = package_json.get("dependencies", {}).get("@ons/design-system")
```

Reply:
Re-implemented the changes in this pattern